File: 1-easy/problems/0482-license-key-formatting/solution.py
def licenseKeyFormatting(S, K):
    cleaned = S.replace('-', '').upper()

    if not cleaned:
        return ""

    result = []
    for i in range(len(cleaned) - 1, -1, -1):
        if len(result) > 0 and len(result) % (K + 1) == K:
            result.append('-')
        result.append(cleaned[i])

    return ''.join(reversed(result))

File: 1-easy/problems/0482-license-key-formatting/solution.cpp
class Solution {
public:
    string licenseKeyFormatting(string S, int K) {
        string cleaned = "";
        for (char c : S) {
            if (c != '-') {
                cleaned += toupper(c);
            }
        }

        if (cleaned.empty()) return "";

        string result = "";
        int count = 0;
        for (int i = cleaned.length() - 1; i >= 0; i--) {
            if (count == K) {
                result = "-" + result;
                count = 0;
            }
            result = cleaned[i] + result;
            count++;
        }

        return result;
    }
};
